### ISSUE TITLE
Random freezes

### ISSUE THREAD

**Issue by DmitryUlyanov** [BVLC/caffe#1807]
They are just exactly as described in https://github.com/BVLC/caffe/issues/1412 thread. The point where it freezes is completely random. It seems like caffe looses gpu somehow (or blocks gpu?) because gpu does not answer after caffe is frozen and before it is killed. 

Particularly  this happens when I try use kernel_size = 1 in convolution layers. So this issue is config dependent.

Upd: it happens at random when I run my other configs, but using kernel_size=1 it happened at the very start of the training process

**Comment by talda** [comment]
The problem described in #1412 is related to the use of LMDB (especially when it is used in more then one training processes in parallel).

I have stopped using the LMDB and these freezes disappeared completely.

**Comment by DmitryUlyanov** [comment]
I am using MemoryData layers and having this issue, tried to switch it to LMDB, had the same problem.

**Comment by nik-ko** [comment]
I'm encountering the same problem using leveldb input on a Tesla K20Xm 5759MiB Driver Version: 331.62
whereas it seemed to work fine on a GeForce GTX 770 4095MiB memory Driver Version: 331.113 i.e. it never froze even when training comprised ~400k iterations.

**Comment by longjon** [comment]
Closing as there have been significant changes to the data code since the main issue here occurred. Please open a new issue if you are still experiencing significant problems with `master` in `DEBUG` mode.
